vassal-app/src/main/java/VASSAL/counters/GlobalCommand.java

```diff
@@ -83,51 +101,303 @@ public void setReportSingle(boolean reportSingle) {
+          GamePiece[] everythingOnMap = map.getPieces();
+
+          if (!target.fastMatchLocation) {
+            // If NOT doing Location fast-matching we do tighter loops (because perf is important during GKCs)
+            if (!target.fastMatchProperty) {
+              // This is the no-fast-matching-at-all version, with "minimum extra overhead" since it's already going to be slow.
+              for (GamePiece pieceOrStack : everythingOnMap) {
+                dispatcher.accept(pieceOrStack);
+              }
+            }
+            else {
+              // This loop is WITH property Fast Match but WITHOUT location Fast Match
+              for (GamePiece pieceOrStack : everythingOnMap) {
+                List<GamePiece> pieceList;
+
+                // We may have an individual piece, or we may have a Stack (or Deck), in which case we need to traverse it.
+                if (pieceOrStack instanceof Stack) {
+                  pieceList = ((Stack) pieceOrStack).asList();
+                }
+                else {
+                  pieceList = Collections.singletonList(pieceOrStack); // Or if really just a single piece.
+                }
+
+                // This will iterate through actual game pieces
+                for (GamePiece gamePiece : pieceList) {
+                  // If a property-based Fast Match is specified, we eliminate non-matchers of that first.
+                  if (!passesPropertyFastMatch(gamePiece)) {
+                    continue;
+                  }
+                  dispatcher.accept(gamePiece);
+                }
+              }
+            }
+          }
+          else {
+            // WITH Location Fast Matching we have some extra steps
+            for (GamePiece pieceOrStack : everythingOnMap) {
+              List<GamePiece> pieceList;
+
+              // We may have an individual piece, or we may have a Stack (or Deck), in which case we need to traverse it.
+              if (pieceOrStack instanceof Stack) {
+                pieceList = ((Stack) pieceOrStack).asList();
+              }
+              else {
+                pieceList = Collections.singletonList(pieceOrStack); // Or if really just a single piece.
+              }
+
+              // This will iterate through actual game pieces
+              for (GamePiece gamePiece : pieceList) {
+                // If a property-based Fast Match is specified, we eliminate non-matchers of that first.
+                if (!passesPropertyFastMatch(gamePiece)) {
+                  continue;
+                }
+
+                // These basic location filters are faster than equivalent filters in the Beanshell expression
+                if ((target.targetType == GlobalCommandTarget.Target.ZONE) || (target.targetType == GlobalCommandTarget.Target.LOCATION) ||
+                  (target.targetType == GlobalCommandTarget.Target.CURZONE) || (target.targetType == GlobalCommandTarget.Target.CURLOC)) {
```

Comment:
I don't think this if statement is needed. All options are covered in the immediately following switch statement. Just add a default: to the switch that does nothing.

Reply:
Oh yeah, I think there was some earlier version of this with an intermediate step that made this make sense. Removed.